### PR TITLE
OSOE-83: Include dot files during 'dotnet pack'

### PR DESCRIPTION
**Pull request by 0liver**
OSOE-83

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -131,7 +131,7 @@ jobs:
         #   a non-packagable project.
         # * -p:IncludeSymbols=true -p:SymbolPackageFormat=snupkg are needed to generate symbol packages:
         #   https://docs.microsoft.com/en-us/nuget/create-packages/symbol-packages-snupkg.
-        run: dotnet pack --output artifacts --configuration Release --no-restore --no-build -p:NuGetBuild=true -p:Version=${{ steps.get_version.outputs.VERSION }} -p:GenerateDocumentationFile=True -p:NoWarn=NU5104 -p:TreatWarningsAsErrors=true -warnaserror -p:WarnOnPackingNonPackableProject=True -p:IncludeSymbols=true -p:SymbolPackageFormat=snupkg
+        run: dotnet pack --output artifacts --configuration Release --no-restore --no-build -p:NuGetBuild=true -p:Version=${{ steps.get_version.outputs.VERSION }} -p:GenerateDocumentationFile=True -p:NoWarn=NU5104 -p:TreatWarningsAsErrors=true -warnaserror -p:WarnOnPackingNonPackableProject=True -p:IncludeSymbols=true -p:SymbolPackageFormat=snupkg -p:NoDefaultExcludes=true
```
Comment (0liver) [inline .github/workflows/publish.yml]: @Piedone The `NoDefaultExcludes` prevents the default exclusion of dot files.

Comment (Piedone) [reply .github/workflows/publish.yml]: Why is this necessary

Comment (0liver) [reply .github/workflows/publish.yml]: Because I have both `.npmrc` and `.stylelintrc` in NE that need to be in the package.

Comment (Piedone) [reply .github/workflows/publish.yml]: This also packs .github folders and .git files.

Comment (0liver) [reply .github/workflows/publish.yml]: Hm, didn't notice that. I thought only those files would be packed that are listed in the project file 🤔

Comment (0liver) [reply .github/workflows/publish.yml]: Where did you see `.git` and `.github` being packed? I can't see any, neither in `Lombiq.Privacy` nor in `NodeJs.Extensions.`

Comment (Piedone) [reply .github/workflows/publish.yml]: This is what the docs say: https://docs.microsoft.com/en-us/nuget/reference/cli-reference/cli-ref-pack#:~:text=Prevents%20default%20exclusion%20of%20NuGet%20package%20files%20and%20files%20and%20folders%20starting%20with%20a%20dot%2C%20such%20as%20.svn%20and%20.gitignore. However, I tried this out with Helpful Extensions now, and indeed Git files are not packed, so the docs are wrong.

However, I also didn't notice any difference without this config, .stylelintrc is just embedded into the DLL under Resources.

Comment (0liver) [reply .github/workflows/publish.yml]: The point is that I need the dot files as files, not in the DLL ;-)

As for the docs, I'd say they are somewhat misleading. What they are trying to say is that by default no dot files will be packed, even if you specify them to be so. You need to pass that flag to be able to include dot files in your package. Which is what we need.

Comment (Piedone) [reply .github/workflows/publish.yml]: I didn't see  .stylelintrc being included as a content file either. Anyway, it doesn't matter too much, if it works in GitHub Actions without including unwanted files then good, and I see from what I can download that it does work (NE and Privacy).

**Review by Piedone — APPROVED** [review]

File: .github/workflows/publish.yml (hunk 1 above)
Comment (0liver) [reply .github/workflows/publish.yml]: Great then!

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -2,10 +2,14 @@ name: publish
...
   workflow_call:
-    inputs: 
+    inputs:
       source:
         required: false
         type: string
+      verbosity:
+        required: false
+        type: string
+        default: minimal
```
Comment (0liver) [inline .github/workflows/publish.yml:12]: I've added this optional verbosity parameter to more easily test the NuGet build and publish for Lombiq.Privacy. You would use like this:

image

Comment (0liver) [reply .github/workflows/publish.yml:12]: I'm inclined to think that the same would work for the `source` parameter, which is currently assigned using an extra action (lines 23ff.)

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -110,32 +114,32 @@ jobs:
...
         #   https://docs.microsoft.com/en-us/nuget/create-packages/symbol-packages-snupkg.
-        run: dotnet pack --output artifacts --configuration Release --no-restore --no-build -p:NuGetBuild=true -p:Version=${{ steps.get_version.outputs.VERSION }} -p:GenerateDocumentationFile=True -p:NoWarn=NU5104 -p:TreatWarningsAsErrors=true -warnaserror -p:WarnOnPackingNonPackableProject=True -p:IncludeSymbols=true -p:SymbolPackageFormat=snupkg
+        run: dotnet pack --output artifacts --configuration Release --no-restore --no-build -p:NuGetBuild=true -p:Version=${{ steps.get_version.outputs.VERSION }} -p:GenerateDocumentationFile=True -p:NoWarn=NU5104 -p:TreatWarningsAsErrors=true -warnaserror -p:WarnOnPackingNonPackableProject=True -p:IncludeSymbols=true -p:SymbolPackageFormat=snupkg -p:NoDefaultExcludes=true --verbosity ${{ inputs.verbosity }}
       - name: Push with dotnet
         run: dotnet nuget push artifacts/*.nupkg --api-key ${{ secrets.API_KEY }} --source ${{ steps.sourceeval.outputs.value }} --skip-duplicate
       - name: Archive artifacts
-        uses: actions/upload-artifact@v1
+        uses: actions/upload-artifact@v3
```
Comment (0liver) [inline .github/workflows/publish.yml:142]: I've updated this action to the latest version, which has a lot of nice features that I used for testing. Nothing breaks here.

File: .github/workflows/publish.yml (hunk 2 above)
Comment (Piedone) [reply .github/workflows/publish.yml:12]: Add this to [the docs](https://github.com/Lombiq/GitHub-Actions#nuget-publish-workflow) too.

**Review by Piedone — CHANGES_REQUESTED** [review]

**Review by Piedone — APPROVED** [review]

**Pull request merged**